lib/subdomain_constraint.rb

```diff
@@ -1,5 +1,5 @@
 class SubdomainConstraint
   def self.matches?(request)
-    Channel.pluck(:permalink).include? request.subdomain
+    User.with_permalink.pluck(:permalink).include? request.subdomain
```

Comment:
Seriously, I think we should just test for some reserverd domains in a constant, if it's not one of the reserved domains we redirect to user#show and let the controller find the user or trigger a 404.
Otherwise this pluck is going to become very memory hungry with time.